pkg/api/pod/util.go

```diff
@@ -23,6 +23,18 @@ import (
 	"k8s.io/kubernetes/pkg/util/intstr"
 )
 
+const (
+	// The annotation value is a string specifying the hostname to be used for the pod e.g 'my-webserver-1'
+	PodHostnameAnnotation = "pod.beta.kubernetes.io/hostname"
+
+	// The annotation value is a string specifying the subdomain e.g. "my-web-service"
+	// If specified, on the the pod itself, "<hostname>.my-web-service.<namespace>.svc.<cluster domain>" would resolve to
+	// the pod's IP.
+	// If there is a headless service named "my-web-service" in the same namespace as the pod, then,
+	// <hostname>.my-web-service.<namespace>.svc.<cluster domain>" would be resolved by the cluster DNS Server.
+	PodSubdomainAnnotation = "pod.beta.kubernetes.io/subdomain"
```

Comment:
To elaborate some more, I am guessing the goal is you want to be able to ping pod1 from pod2 by its FQDN.

If I annotated pod 1 with the hostname "pet1", and it got a pod IP of 172.17.0.4, pod 2 can do the following:

```
$ ping 172.17.0.4
$ ping 172-17-0-4.default.pod.cluster.local
```

Why do we need a headless service and can we not just create a record based on pod1 host name?

So in this case, pod 2 could also do:

```
$ ping pet1.default.pod.cluster.local
```

Is that enough for your use case?